src/core_init_atmosphere/mpas_kd_tree.F

```diff
@@ -158,6 +161,47 @@ function mpas_kd_construct(points, ndims) result(tree)
 
    end function mpas_kd_construct
 
+   !***********************************************************************
+   !
+   !  routine break_tie
+   !
+   !> \brief   Break a tie for two n-dim points
+   !> \author  Miles A. Curry
+   !> \date    07/07/20
+   !> \details
+   !> Compare 1..n dimensions of p1 and p2 and return -1 if p1(n) is less then
+   !> p2(n) and return 1 if p1(n) is greater than p2(n). If p1(n) and p2(n) are
+   !> equal than the some comparison will be done on p1(n+1) and p2(n+1). If 
+   !> p1(:) and p2(:) are equal across all n (i.e. they are the exactly the same 
+   !> point), then 0 will be returned.
+   !
+   !-----------------------------------------------------------------------
+   function break_tie(p1, p2) result(tie)
+
+      implicit none
+
+      ! Input Variables
+      type (mpas_kd_type), target, intent(in) :: p1
+      type (mpas_kd_type), target, intent(in) :: p2
```

Comment:
Is the `target` attribute needed for `p1` and `p2`?

Reply:
It is not, I always forget that Fortran is pass by reference.